.claude/hooks/python-design-gate.sh

```
@@ -0,0 +1,47 @@
+
+# CRITICAL: never loop. stop_hook_active=true = Claude is already forced-continuing.
+if [ "$(echo "$INPUT" | jq -r '.stop_hook_active')" = "true" ]; then
+  exit 0
+fi
+
+# 1. Does this diff touch Python files at all?
+CHANGED_PY=$(git diff --name-only HEAD 2>/dev/null | grep '\.py$' || true)
+[ -z "$CHANGED_PY" ] && exit 0   # nothing to gate
+
+# 2. SHA of the Python-relevant diff
+DIFF_SHA=$(git diff HEAD -- $CHANGED_PY 2>/dev/null | shasum -a 256 | cut -d' ' -f1)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)